corehq/apps/hqcase/tests/test_case_api_permissions.py

```diff
@@ -0,0 +1,213 @@
+            self.assertEqual(response.status_code, 200)
+
+    def test_case_api_list_requires_access_all_locations(self):
+        permissions = {
+            'edit_data': True,
+            'access_api': True,
+            'access_all_locations': False,
+        }
+        with get_web_user(
+            self.domain,
+            self.user_location,
+            permissions,
+            self.client,
+        ):
+            url = reverse('case_api', args=(self.domain,))
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, 403)
+
+    def test_case_api_bulk_fetch_happy_path(self):
+        permissions = {
+            'edit_data': True,
+            'access_api': True,
+        }
+        with get_web_user(
+            self.domain,
+            self.user_location,
+            permissions,
+            self.client,
+        ):
+            url = reverse('case_api_bulk_fetch', args=(self.domain,))
+            response = self.client.post(
+                url,
+                {'case_ids': self.case_ids},
+                content_type='application/json',
+            )
+            self.assertEqual(response.status_code, 200)
+
+    def test_case_api_bulk_fetch_requires_access_all_locations(self):
+        permissions = {
+            'edit_data': True,
+            'access_api': True,
+            'access_all_locations': False,
+        }
+        with get_web_user(
+            self.domain,
+            self.user_location,
+            permissions,
+            self.client,
+        ):
+            url = reverse('case_api_bulk_fetch', args=(self.domain,))
+            response = self.client.post(
+                url,
+                {'case_ids': self.case_ids},
+                content_type='application/json',
+            )
+            self.assertEqual(response.status_code, 403)
+
+    def test_get_list_access_all_locations(self):
+        permissions = {
+            'edit_data': True,
+            'access_api': True,
+            'access_all_locations': True,
+        }
+        with get_web_user(
+            self.domain,
+            self.user_location,
+            permissions,
+            self.client,
+        ) as web_user:
+
+            result = get_list(self.domain, web_user, params={})
+            self.assertEqual(result['matching_records'], 2)
+            self.assertEqual(
+                {c['case_name'] for c in result['cases']},
+                {'Joe', 'Cyril'},
+            )
+
+    def test_get_list_location_restricted(self):
+        permissions = {
+            'edit_data': True,
+            'access_api': True,
+            'access_all_locations': False,
+        }
+
+        with sync_users_to_es():
+            with get_web_user(
+                self.domain,
+                self.user_location,
+                permissions,
+                self.client,
+            ) as web_user:
+                get_es_new().indices.refresh(USER_INDEX)
```

Comment:
This is done differently with the new Document Adapter layer:

```suggestion
                manager.index_refresh(user_adapter.index_name)
```